Modules/Loadable/Markups/MRML/vtkMRMLMarkupsROINode.cxx

```diff
@@ -138,6 +138,62 @@ void vtkMRMLMarkupsROINode::CreateDefaultDisplayNodes()
   this->SetAndObserveDisplayNodeID(dispNode->GetID());
 }
 
+//---------------------------------------------------------------------------
+void vtkMRMLMarkupsROINode::ApplyTransform(vtkAbstractTransform* transform)
+{
+  if (!transform)
+    {
+    vtkErrorMacro("ApplyTransform: Invalid transform");
+    return;
+    }
+
+  MRMLNodeModifyBlocker blocker(this);
+
+  bool wasUpdatingControlPointsFromROI = this->IsUpdatingControlPointsFromROI;
+  this->IsUpdatingControlPointsFromROI = true;
+
+  bool wasUpdatingROIFromControlPoints = this->IsUpdatingROIFromControlPoints;
+  this->IsUpdatingROIFromControlPoints = true;
+
+  double center_Local[3] = { 0.0, 0.0, 0.0 };
+  this->GetCenter(center_Local);
+
+  double xAxis_Local[3] = { 1.0, 0.0, 0.0 };
+  this->GetXAxisLocal(xAxis_Local);
+  transform->TransformVectorAtPoint(center_Local, xAxis_Local, xAxis_Local);
+  double xAxisSize = vtkMath::Normalize(xAxis_Local);
+
+  double yAxis_Local[3] = { 0.0, 1.0, 0.0 };
+  this->GetYAxisLocal(yAxis_Local);
```

Comment:
Let me know if the GenerateOrthogonalMatrix simplifies the process enough. Internally it still does some vector-by-vector operations, but now it's all collected in one place. I'm not sure if there's a way around it since the applied transform may not be linear.